src/main/java/sawtooth/sdk/client/signing/PublicKey.java

```diff
@@ -0,0 +1,28 @@
+package sawtooth.sdk.client.signing;
```

Comment:
Curious question and a thought: (sorry for being verbose)
I see this is the way traits are created in Rust, that is to have PrivateKey and PublicKey as separate interfaces. In case of Java since one class can implement multiple interfaces it'll give impression that ABCAlgoKey can implement both of these.
Would it be nice to have single interface Key, PrivateKey and PublicKey may be abstract implementation of it?

Reply:
IMHO, there will code that will only have access to the PublicKey methods, due to the role it is executing (like on clients, consumers, etc).

So, keeping an interface with signatures only referring to PublicKey is a must do, but a "server" side that implements both interfaces is a possibility.

On top of that, on secp256k1 (and other asymmetric keys as well)   the Private Key can generate the Public Key for you, anytime.

Reply:
Agree, thanks for more details